tests/integration_tests.py

```diff
@@ -54,15 +55,28 @@ def test_chunked_uploads(local_grand_challenge):
         os.path.dirname(os.path.realpath(__file__)), "testdata", "rnddata"
     )
     # admin
-    c_admin = Client(token=ADMIN_TOKEN, base_url=local_grand_challenge, verify=False,)
+    c_admin = Client(token=ADMIN_TOKEN, base_url=local_grand_challenge, verify=False)
     c_admin.chunked_uploads.send(file_to_upload)
     assert c_admin(path="chunked-uploads/")["count"] == 1
 
     # retina
-    c_retina = Client(token=RETINA_TOKEN, base_url=local_grand_challenge, verify=False,)
+    c_retina = Client(token=RETINA_TOKEN, base_url=local_grand_challenge, verify=False)
     c_retina.chunked_uploads.send(file_to_upload)
     assert c_retina(path="chunked-uploads/")["count"] == 1
 
     c = Client(token="whatever")
     with pytest.raises(HTTPError):
         c.chunked_uploads.send(file_to_upload)
+
+
+def test_run_external_algorithm(local_grand_challenge):
+    c = Client(base_url=local_grand_challenge, verify=False, token=ALGORITHMUSER_TOKEN)
+
+    image_to_upload = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)), "testdata", "image10x10x101.mha"
+    )
+    assert c.algorithm_jobs.list()["count"] == 0
+    us_pk = c.run_external_algorithm("Test Algorithm", image_to_upload)
+    # Need to login again to access updated information
```

Comment:
Why do you need to login again? That shouldn't be necessary.

Reply:
Does not work otherwise locally. I can see the job if I refresh the page and/or after I login. I just added a wait condition https://github.com/DIAGNijmegen/rse-gcapi/blob/run_external_algo_i31/gcapi/gcapi.py#L502
I am not sure if this is going to help.

Reply:
I am working on it.

Reply:
Fixed